zookeeper/src/main/java/com/linecorp/armeria/client/zookeeper/ZookeeperDiscoverySpec.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.zookeeper;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.server.zookeeper.ZookeeperRegistrationSpec;
+
+/**
+ * A discovery specification for {@link ZooKeeperEndpointGroup}. The specification is used for finding
+ * and decoding the registered instances into {@link Endpoint}s.
+ *
+ * @see ZookeeperRegistrationSpec
+ */
+public interface ZookeeperDiscoverySpec {
+
+    /**
+     * Returns a {@link ZookeeperDiscoverySpec} that is compatible with
+     * <a href="https://curator.apache.org/curator-x-discovery/index.html">Curator-X-Discovery</a>.
+     * This is also, compatible with
+     * <a href="https://cloud.spring.io/spring-cloud-zookeeper/reference/html/">Spring Cloud Zookeeper</a>.
+     */
+    static ZookeeperDiscoverySpec ofCuratorX(String serviceName) {
```

Comment:
Which one sounds better, `ofCurator()`, `forCurator()` or just `curator()`?
For example, we don't use any prefix in `Backoff`, except for `ofDefault()`.